snmp/README.md

```diff
@@ -19,12 +19,66 @@ The SNMP check is included in the [Datadog Agent][1] package. No additional inst
 
-Edit the subnet and SNMP version in the `snmp.d/conf.yaml` file in the `conf.d/` folder at the root of your [Agent's configuration directory][2]. See the [sample snmp.d/conf.yaml][3] for all available configuration options.
+The Datadog SNMP check supports collecting metrics from individual devices, or auto-discovering devices on entire subnets.
+
+Which collection strategy to choose mainly depends on how many devices are present on your network, and how dynamic the network is (i.e. how frequently devices are added or removed):
+
+- For small and mostly static networks, see [Monitoring individual devices](#monitoring-individual-devices).
+- For larger and/or dynamic networks, see [Autodiscovery](#autodiscovery).
+
+Regardless of the collection strategy, you can leverage Datadog's [`sysObjectID` mapped device profiles](#sysobjectid-mapped-device-profiles) to automatically collect relevant metrics from your devices.
+
+#### Monitoring individual devices
+
+The easiest way to get started with the SNMP integration is to point it at the IP address of an SNMP device.
+
+For a minimal setup, we'll use the `generic-router` profile. It collects generic metrics exposed by most network devices (such as ones in `IF-MIB`, `IP-MIB`, etc).
+
+For SNMPv2 devices:
+
+1. Configure an instance specifying i) the IP address of the device, ii) the `community` string of the device, and iii) the `generic-router` profile:
+
+    ```yaml
+    instances:
+      - ip_address: "<IP_ADDRESS>"
+        community: "<COMMUNITY>"
+        profile: generic-router
+    ```
+
+2. [Restart the Agent][31].
+
+For SNMPv3 devices:
+
+1. Configure an instance specifying i) the IP address of the device, ii) SNMPv3 credentials of the device, i.e. `user` and any of `auth_protocol`, `auth_key`, `priv_protocol`, and `priv_key` (as appropriate for your device), and iii) the `generic-router` profile:
+
+    ```yaml
+    instances:
+      - ip_address: "<IP_ADDRESS>"
+        user: "<USER>"
+        ## Configure these as appropriate
+        # authProtocol: SHA
+        # authKey: "<AUTH_KEY>"
+        # privProtocol: AES
+        # privKey: "<PRIV_KEY>"
+        profile: generic-router
+    ```
+
+2. [Restart the Agent][31].
+
+Once metrics are reporting, you can expand your setup in various ways:
+
+* Use a more specific [`sysObjectID` mapped device profile](#sysobjectid-mapped-device-profiles) to collect metrics specific to your device.
+* Add more instances to collect metrics from more devices on your network.
+* Consider using [Autodiscovery](#autodiscovery) if you need to collect metrics from lots of devices across a dynamic network.
 
 #### Autodiscovery
 
+As the number of devices to collect metrics from grows, individual device configuration can become tedious. You may also want to configure your SNMP integration so that it automatically discovers new devices added to your network, or stops monitoring devices that were removed from the network.
```

Comment:
Instead of calling out that our integration might become tedious, I think we can phrase this in a more positive way -- something along the lines of, 

You can also use autodiscovery to automatically discover all of the devices on your network. Autodiscovery will poll each IP on a given subnet, and check for a response from the SNMP port. The Datadog Agent will then lookup the sysObjectId for the discovered device, and map that to Datadog's out-of-the-box device profiles [link to the profiles section can go here], which consist of predefined list of metrics to be collected for that specific type of device.